File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        int count=0;
        unordered_map<int,int>mp;

        for (int i=0;i<arr.size();i++) {
            int rem=((arr[i]%k)+k)%k;
            int need=(k-rem)%k;

            if (mp.find(need)!=mp.end()&&mp[need]>0){
                count++;
                mp[need]--;
            } else {
                mp[rem]++;
            }
        }

        return count >= arr.size() / 2;
    }
};

File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        int count=0;
        Map<Integer,Integer>mp=new HashMap<>();
        for(int i=0;i<arr.length;i++){
            int rem=((arr[i]%k)+k)%k;
            int need=(k-rem)%k;
            if(mp.containsKey(need) && mp.get(need)>0){
                count++;
                  mp.put(need,mp.getOrDefault(need,0)-1);
            }
            else{
                mp.put(rem,mp.getOrDefault(rem,0)+1);
            }
        }
        if(count>=arr.length/2){
            return true;
        }
        return false;
    }
}